docs/apis/batch/libs/gelly.md

```diff
@@ -1734,3 +1734,547 @@ vertex represents a group of vertices and each edge represents a group of edges
+    <line x1="190" y1="120" x2="350" y2="120" />
+    <line x1="190" y1="200" x2="350" y2="200" />
+    <line x1="190" y1="120" x2="190" y2="200" />
+    <line x1="350" y1="120" x2="350" y2="200" />
+
+    <line x1="30" y1="40" x2="510" y2="40" />
+    <line x1="30" y1="280" x2="510" y2="280" />
+    <line x1="30" y1="40" x2="30" y2="280" />
+    <line x1="510" y1="40" x2="510" y2="280" />
+
+    <line x1="190" y1="120" x2="30" y2="40" />
+    <line x1="350" y1="120" x2="510" y2="40" />
+    <line x1="190" y1="200" x2="30" y2="280" />
+    <line x1="350" y1="200" x2="510" y2="280" />
+
+    <circle cx="190" cy="120" r="20" />
+    <text x="190" y="120">0</text>
+
+    <circle cx="350" cy="120" r="20" />
+    <text x="350" y="120">1</text>
+
+    <circle cx="190" cy="200" r="20" />
+    <text x="190" y="200">2</text>
+
+    <circle cx="350" cy="200" r="20" />
+    <text x="350" y="200">3</text>
+
+    <circle cx="30" cy="40" r="20" />
+    <text x="30" y="40">4</text>
+
+    <circle cx="510" cy="40" r="20" />
+    <text x="510" y="40">5</text>
+
+    <circle cx="30" cy="280" r="20" />
+    <text x="30" y="280">6</text>
+
+    <circle cx="510" cy="280" r="20" />
+    <text x="510" y="280">7</text>
+</svg>
+
+### Path Graph
+
+An undirected Graph where all edges form a single path.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+Graph<LongValue,NullValue,NullValue> graph = new PathGraph(env, 5)
+    .generate();
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+import org.apache.flink.api.scala._
+import org.apache.flink.graph.generator.PathGraph
+
+val env: ExecutionEnvironment = ExecutionEnvironment.getExecutionEnvironment
+
+val graph = new PathGraph(env.getJavaEnv, 5).generate()
+{% endhighlight %}
+</div>
+</div>
+
+<svg class="graph" width="540" height="80"
+    xmlns="http://www.w3.org/2000/svg"
+    xmlns:xlink="http://www.w3.org/1999/xlink">
+
+    <line x1="30" y1="40" x2="510" y2="40" />
+
+    <circle cx="30" cy="40" r="20" />
+    <text x="30" y="40">0</text>
+
+    <circle cx="150" cy="40" r="20" />
+    <text x="150" y="40">1</text>
+
+    <circle cx="270" cy="40" r="20" />
+    <text x="270" y="40">2</text>
+
+    <circle cx="390" cy="40" r="20" />
+    <text x="390" y="40">3</text>
+
+    <circle cx="510" cy="40" r="20" />
+    <text x="510" y="40">4</text>
+</svg>
+
+### RMat Graph
+
+A directed or undirected power-law graph generated using the [Recursive Matrix (R-Mat)]
+(http://www.cs.cmu.edu/~christos/PUBLICATIONS/siam04.pdf) model. RMat is a stochastic
+generator configured with a source of randomness.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+RandomGenerableFactory<JDKRandomGenerator> rnd = new JDKRandomGeneratorFactory();
+
```

Comment:
Can we explain what is the `JDKRandomGenerator ` and what parameters the `RMatGraph` takes?